ledger/src/blockstore.rs

```diff
@@ -170,6 +174,50 @@ impl<T> AsRef<T> for WorkingEntry<T> {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct LastFECSetCheckResults {
+    block_id: Option<Hash>,
+    is_retransmitter_signed: bool,
+}
+
+impl LastFECSetCheckResults {
+    pub fn report_metrics(&self, slot: Slot, bank_hash: Hash, cluster_type: ClusterType) {
+        if self.block_id.is_none() {
+            datapoint_warn!(
+                "incomplete_final_fec_set",
+                ("slot", slot, i64),
+                ("bank_hash", bank_hash.to_string(), String)
+            );
+        }
+        // These metrics are expensive to send because hash does not compress well.
+        // Only send these metrics when we are sure the appropriate shred format is being sent
+        if !self.is_retransmitter_signed && shred::should_chain_merkle_shreds(slot, cluster_type) {
+            datapoint_warn!(
+                "invalid_retransmitter_signature_final_fec_set",
+                ("slot", slot, i64),
+                ("bank_hash", bank_hash.to_string(), String)
+            );
+        }
```

Comment:
As in the other comment, maybe we should remove this metric entirely for now and add it back once chained merkle shreds are fully rolled out.